osquery/distributed/distributed.cpp

```diff
@@ -155,205 +275,200 @@ Status Distributed::flushCompleted() {
-                                       JSON& doc,
-                                       rj::Value& obj) {
-  auto request_obj = doc.getObject();
-  auto s = serializeDistributedQueryRequest(r.request, doc, request_obj);
-  if (!s.ok()) {
-    return s;
+  if (!lastQueryId.empty()) {
+    LOG(WARNING) << "distributed worker was interrupted running query id "
+                 << lastQueryId;
   }
 
-  auto results_arr = doc.getArray();
-  s = serializeQueryData(r.results, r.columns, doc, results_arr);
-  if (!s.ok()) {
-    return s;
+  // load queries
+
+  populateResultState(doc, Status());
+
+  // mark with INTERRUPTED status
+
+  for (auto& item : results_) {
+    item.status = Status(DQ_INTERRUPTED_STATUS, "INTERRUPTED");
   }
 
-  doc.add("request", request_obj);
-  doc.add("results", results_arr);
-  return Status();
+  // send response
+
+  flushCompleted();
 }
 
-Status serializeDistributedQueryResultJSON(const DistributedQueryResult& r,
-                                           std::string& json) {
+Status Distributed::acceptWork(const std::string& work) {
   auto doc = JSON::newObject();
-  auto s = serializeDistributedQueryResult(r, doc, doc.doc());
-  if (!s.ok()) {
-    return s;
+  if (!doc.fromString(work) || !doc.doc().IsObject()) {
+    return Status(1, "Error Parsing JSON");
   }
 
-  return doc.toString(json);
-}
+  Status discoveryStatus = passesDiscovery(doc);
 
-Status deserializeDistributedQueryResult(const rj::Value& obj,
-                                         DistributedQueryResult& r) {
-  DistributedQueryRequest request;
-  auto s = deserializeDistributedQueryRequest(obj["request"], request);
-  if (!s.ok()) {
-    return s;
-  }
+  if (doc.doc().HasMember("queries")) {
+    const auto& queries = doc.doc()["queries"];
 
-  QueryData results;
-  s = deserializeQueryData(obj["results"], results);
-  if (!s.ok()) {
-    return s;
-  }
+    if (!queries.IsObject()) {
+      return Status(1, "Format error: Distributed 'queries' is not object");
+    }
 
-  r.request = request;
-  r.results = results;
+    Status status = populateResultState(doc, discoveryStatus);
+    if (status.ok()) {
+      // good to go. Save work to DB in case watchdog kills worker
+      setDatabaseValue(kPersistentSettings, "distributed_work", work);
+    } else {
+      return status; // parsing failed
+    }
+  }
 
+  if (doc.doc().HasMember("accelerate")) {
+    const auto& accelerate = doc.doc()["accelerate"];
+    if (accelerate.IsInt() && accelerate.GetInt() > 0 &&
+        accelerate.GetInt() < MAX_ACCEL_DURATION_SEC) {
+      auto duration = accelerate.GetInt();
+      LOG(INFO) << "Accelerating distributed query checkins for " << duration
+                << " seconds";
+      setDatabaseValue(kPersistentSettings,
+                       "distributed_accelerate_checkins_expire",
+                       std::to_string(getUnixTime() + duration));
+    } else {
+      VLOG(1) << "Invalid argument for accelerate. must be int > 0 < MAX";
```

Comment:
You should log what that value is

Reply:
I would be afraid of an exception.  accelerate is a JSON abstracted type that is expected int, but could be any type. is there a to_s() equivalent?